packages/cli/src/controller/scaffoldgen-controller.ts

```diff
@@ -0,0 +1,232 @@
+      kind: 'ethereum/TransactionHandler',
+      filter: {
+        function: fn.method,
+      },
+    };
+    formattedHandlers.push(handler);
+  });
+
+  userInput.events.map((event) => {
+    const handler: HandlerType = {
+      handler: `handle${upperFirst(event.name)}_${abiName}Log`,
+      kind: 'ethereum/LogHandler',
+      filter: {
+        topics: [event.method],
+      },
+    };
+    formattedHandlers.push(handler);
+  });
+
+  return {
+    kind: 'ethereum/Runtime',
+    startBlock: userInput.startBlock,
+    options: {
+      abi: abiName,
+      address: userInput.address,
+    },
+    assets: {
+      [abiName]: {
+        file: userInput.abiPath,
+      },
+    },
+    mapping: {
+      file: DEFAULT_HANDLER_BUILD_PATH,
+      handlers: formattedHandlers,
+    },
+  };
+}
+
+export async function generateManifest(projectPath: string, manifestPath: string, userInput: UserInput): Promise<void> {
+  try {
+    const existingManifest = (await fs.promises.readFile(path.join(projectPath, manifestPath), 'utf8')) as any;
+    const existingManifestData = parseDocument(existingManifest);
+    const clonedExistingManifestData = existingManifestData.clone();
+
+    const existingDatasource = existingManifestData.get('dataSources') as any;
+
+    // Should be for every ABI
+
+    const newDataSourcesData = existingDatasource.toJSON().concat(...[constructDatasources(userInput)]);
+    clonedExistingManifestData.set('dataSources', newDataSourcesData);
+
+    // load yaml
+    await fs.promises.writeFile(path.join(projectPath, manifestPath), clonedExistingManifestData.toString(), 'utf8');
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+export interface handlerPropType {
+  name: string;
+  argName: string;
+  argType: string;
+}
+
+export interface abiPropType {
+  name: string;
+  handlers: handlerPropType[];
+}
+
+export function constructHandlerProps(methods: [SelectedMethod[], SelectedMethod[]], abiName: string): abiPropType {
+  const handlers: handlerPropType[] = [];
+  const [events, functions] = methods;
+
+  functions.map((fn) => {
+    const fnProp: handlerPropType = {
+      name: `handle${upperFirst(fn.name)}`,
+      argName: 'tx',
+      argType: `${upperFirst(fn.name)}Transaction`,
+    };
+    handlers.push(fnProp);
+  });
+
+  events.map((event) => {
+    const fnProp: handlerPropType = {
+      name: `handle${upperFirst(event.name)}`,
+      argName: 'log',
+      argType: `${upperFirst(event.name)}Log`,
+    };
+    handlers.push(fnProp);
+  });
+
+  return {
+    name: abiName,
+    handlers: handlers,
+  };
+}
+
+export async function generateHandlers(
+  selectedMethods: [SelectedMethod[], SelectedMethod[]],
+  projectPath: string,
+  abiPath: string
+): Promise<void> {
+  const abiProps = constructHandlerProps(selectedMethods, parseContractPath(abiPath).name);
+
+  try {
+    await renderTemplate(
+      SCAFFOLD_HANDLER_TEMPLATE_PATH,
+      path.join(projectPath, ROOT_MAPPING_DIR, 'mappingHandlers.ts'),
```

Comment:
I think we should create a new file with the name of the abi/contract. It's quite common to add multiple abis to a project. We don't want to overwrite their existing `mappingHandlers` file either. 

I think we should also check if the file exists. Then we could do a couple of things, 1) nothing, 2) create a new file and they can copy what they wish, 3) throw an error. We don't want to wipe their functions if they already exist. 

Then we can append the new import to `index.ts`